sqlglot/parser.py

```diff
@@ -2100,7 +2100,9 @@ def _parse_column_constraint(self):
         elif self._match(TokenType.DEFAULT):
             kind = self.expression(exp.DefaultColumnConstraint, this=self._parse_conjunction())
         elif self._match_pair(TokenType.NOT, TokenType.NULL):
-            kind = exp.NotNullColumnConstraint()
+            kind = exp.NotNullColumnConstraint(allow_null=False)
+        elif self._match(TokenType.NULL):
+            kind = exp.NotNullColumnConstraint(allow_null=True)
```

Comment:
for some reason this didn't work. I thought it might be because allow_null returns True/None instead of True/False but accounting for that didn't solve things.

Reply:
That's fine